golem-worker-executor-base/tests/api.rs

```diff
@@ -1642,23 +1651,21 @@ async fn long_running_poll_loop_works_as_expected(
         .await
         .unwrap();
 
-    sleep(Duration::from_secs(2)).await;
-
-    let (status1, _) = executor.get_worker_metadata(&worker_id).await.unwrap();
+    executor
+        .wait_for_status(&worker_id, WorkerStatus::Running, Duration::from_secs(5))
```

Comment:
Same here, I would propose to just poll forever for the expected status and move operational matters such as timeouts fully into test framework configuration.